runtime/oti/xhelpers.m4

```diff
@@ -43,8 +43,76 @@ ifdef({ASM_J9VM_ENV_DATA64},{
 	LABEL(skip_vzu{}VZU_COUNT):
 })
 
+dnl for(<symbol>=<start>; <symbol> <= <end>; ++<symbol>) { <expr> }
+dnl $1 = symbol name
+dnl $2 = starting value
+dnl $3 = ending value
+dnl $4 = expression
+define({forloop},
+	{define({$1}, {$2})$4
+	ifelse({$2}, {$3}, {},{$0({$1}, incr({$2}), {$3}, {$4})})})
```

Comment:
My earlier comments have not been address; see https://github.com/eclipse-openj9/openj9/pull/22692/#discussion_r2407106460.